src/glyph.ts

```diff
@@ -5,20 +5,94 @@ import { Flow } from './tables';
 import { Element } from './element';
 import { BoundingBoxComputation } from './boundingboxcomputation';
 import { BoundingBox } from './boundingbox';
+import { Font } from './smufl';
+import { RenderContext, TypeProps } from './types/common';
+import { Stave } from './stave';
+import { Stem } from './stem';
+
+export interface DurationCode {
+  common: TypeProps;
+  type: Record<string, TypeProps>;
+}
+
+export interface GlyphProps {
+  code_head: string;
+  dot_shiftY: number;
+  position: string;
+  rest: boolean;
+  line_below: number;
+  line_above: number;
+  stem_up_extension: number;
+  stem_down_extension: number;
+  stem: Stem;
+  code: string;
+  code_flag_upstem: string;
+  code_flag_downstem: string;
+  flag: boolean;
+  width: number;
+  text: string;
+  tabnote_stem_down_extension: number;
+  tabnote_stem_up_extension: number;
+  beam_count: number;
+  duration_codes: Record<string, DurationCode>;
+  validTypes: Record<string, string>;
+  shift_y: number;
+
+  getWidth(a?: number): number;
+
+  getMetrics(): GlyphMetrics;
+}
+export interface GlyphOptions {
+  fontStack: Font[];
+  category?: string;
+}
+export interface GlyphMetrics {
+  width?: number;
+  height?: number;
+  x_min: number;
+  x_max: number;
+  x_shift: number;
+  y_shift: number;
+  scale: number;
+  ha?: number;
+  outline: string[];
+  font?: Font;
+}
+
+export interface FontGlyph {
```

Comment:
Is there a way to share this state? e.g., one derives from the other? (or one contains the other.)

Reply:
@0xfe if you want to clarify the font topic I suggest to include the Font class in this PR, it is not very big and it would be the best way to check that both classes interact fine together. I would rename `smufl.js`to `font.ts`. Is that fine?

Reply:
I have gone for it :)

Reply:
Fonts types simplified

Reply:
This is way better, thanks!